tests/validation/tests/v3_api/resource/terraform/charts/aws/main.tf

```diff
@@ -0,0 +1,181 @@
+#Cluster creation with v1 charts installed
+terraform {
+  required_version = ">= 0.14"
+  required_providers {
+    rancher2 = {
+      source = "rancher/rancher2"
+    }
+
+  }
+}
+
+provider "rancher2" {
+  api_url   = var.rancher_api_url
+  token_key = var.rancher_token_key
+}
+
+resource "rancher2_cluster" "custom" {
+  name = "aws-tf-charts"
+  description = "Rancher2 TF Custom Cluster"
+  rke_config {
+    network {
+      plugin = "canal"
+    }
+  }  
+  enable_cluster_monitoring = true
+  cluster_monitoring_input {
+    answers = {
+      "exporter-kubelets.https" = true
+      "exporter-node.enabled" = true
+      "exporter-node.ports.metrics.port" = 9796
+      "exporter-node.resources.limits.cpu" = "200m"
+      "exporter-node.resources.limits.memory" = "200Mi"
+      "grafana.persistence.enabled" = false
+      "grafana.persistence.size" = "10Gi"
+      "grafana.persistence.storageClass" = "default"
+      "operator.resources.limits.memory" = "500Mi"
+      "prometheus.persistence.enabled" = "false"
+      "prometheus.persistence.size" = "50Gi"
+      "prometheus.persistence.storageClass" = "default"
+      "prometheus.persistent.useReleaseName" = "true"
+      "prometheus.resources.core.limits.cpu" = "1000m",
+      "prometheus.resources.core.limits.memory" = "1500Mi"
+      "prometheus.resources.core.requests.cpu" = "750m"
+      "prometheus.resources.core.requests.memory" = "750Mi"
+      "prometheus.retention" = "12h"
+    }
+  }
+}
+
+resource "rancher2_node_template" "node-template" {
+  name = "tf-aws-node-template"
+  description = "TF AWS node template"
+
+  amazonec2_config {
+    access_key = var.aws_access_key
+    secret_key = var.aws_secret_key
+    ami =  var.ami
+    region = var.aws_region
+    security_group = var.security_groups
+    subnet_id = var.subnet
+    vpc_id = var.vpc_id
+    zone = var.zone
+    root_size = var.root_size
+    instance_type = var.instance_type
+  }
+}
+
+resource "rancher2_node_pool" "node-pool" {
+  cluster_id =  rancher2_cluster.custom.id
+  name = "tf-node-pool"
+  hostname_prefix = var.hostname_prefix
+  node_template_id = rancher2_node_template.node-template.id
+  quantity = 3
```

Comment:
Would `quantity` be more useful parameterized? Also usually for manual testing we do 1 Control Plane, 1 etcd, 3 workers.
If we do the 1-1-3 route, we could parameterize the workers node pool `quantity` and keep CP and etcd as `quantity` 1.

But I am fine if that's not a hard requirement for this.

Reply:
Defaults set to 1 CP, 1 etcd and 3 workers, exposed the parameterized worker_quantity to the Jenkins job